tests/test_version.py

```diff
@@ -5,26 +5,40 @@
 
 def test_version():
     ver = get_git_version(abbrev=7)
-    # Versions should comply with semver, which is where this regex is from
-    # see semver.org and github.com/python-semver/python-semver
+    # Versions should comply with PEP440, which is where this regex is from
+    # see https://peps.python.org/pep-0440/ and regex here:
+    # https://github.com/pypa/packaging/blob/b5f0efdf3986725c2a45437e03d0ee5939c21192/packaging/version.py#L159 # noqa
     VER_RE = re.compile(
         r"""
-            ^
-            (?P<major>0|[1-9]\d*)
-            \.
-            (?P<minor>0|[1-9]\d*)
-            \.
-            (?P<patch>0|[1-9]\d*)
-            (?:-(?P<prerelease>
-                (?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)
-                (?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*
-            ))?
-            (?:\+(?P<build>
-                [0-9a-zA-Z-]+
-                (?:\.[0-9a-zA-Z-]+)*
-            ))?
-            $
-            """,
+        v?
```

Comment:
oh gosh you had to rewrite this regexp :scream: ?

Reply:
noooo it's nicked from the python `packaging` package thank god